src/main/java/org/kairosdb/core/health/HealthCheckResource.java

```diff
@@ -18,98 +19,98 @@
-	@Path("status")
-	public Response corsPreflightStatus(@HeaderParam("Access-Control-Request-Headers") String requestHeaders,
-			@HeaderParam("Access-Control-Request-Method") String requestMethod)
-	{
-		Response.ResponseBuilder responseBuilder = MetricsResource.getCorsPreflightResponseBuilder(requestHeaders, requestMethod);
-		return (responseBuilder.build());
-	}
-
-
-	/**
-	 * Returns the status of each health check.
-	 * @return 200
-	 */
-	@GET
-	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
-	@Path("status")
-	public Response status()
-	{
-		List<String> messages = new ArrayList<String>();
-		for (HealthStatus healthCheck : m_healthCheckService.getChecks())
-		{
-			HealthCheck.Result result = healthCheck.execute();
-			if (result.isHealthy())
-			{
-				messages.add(healthCheck.getName() + ": OK");
-			}
-			else
-			{
-				messages.add(healthCheck.getName() + ": FAIL");
-			}
-		}
-
-		GenericEntity<List<String>> entity = new GenericEntity<List<String>>(messages) {};
-		return setHeaders(Response.ok(entity)).build();
-	}
-
-	private Response.ResponseBuilder setHeaders(Response.ResponseBuilder responseBuilder)
-	{
-		responseBuilder.header("Access-Control-Allow-Origin", "*");
-		responseBuilder.header("Pragma", "no-cache");
-		responseBuilder.header("Cache-Control", "no-cache");
-		responseBuilder.header("Expires", 0);
-
-		return (responseBuilder);
-	}
+public class HealthCheckResource {
+    private final HealthCheckService m_healthCheckService;
+    private HealthCheckMetricProvider healthCheckMetricProvider;
+
+    @Inject
+    @Named("kairosdb.health.healthyResponseCode")
+    private int m_healthyResponse = Response.Status.NO_CONTENT.getStatusCode();
+
+    @Inject
+    public HealthCheckResource(HealthCheckService healthCheckService,
+                               HealthCheckMetricProvider healthCheckMetricProvider) {
+        m_healthCheckService = checkNotNull(healthCheckService);
+        this.healthCheckMetricProvider = healthCheckMetricProvider;
+    }
+
+    @OPTIONS
+    @Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
+    @Path("check")
+    public Response corsPreflightCheck(@HeaderParam("Access-Control-Request-Headers") String requestHeaders,
+                                       @HeaderParam("Access-Control-Request-Method") String requestMethod) {
+        Response.ResponseBuilder responseBuilder = MetricsResource.getCorsPreflightResponseBuilder(requestHeaders, requestMethod);
+        return (responseBuilder.build());
+    }
+
+    /**
+     * Health check
+     *
+     * @return 204 if healthy otherwise 500
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
+    @Path("check")
+    public Response check() {
+        for (HealthStatus healthCheck : m_healthCheckService.getChecks()) {
+            HealthCheck.Result result = healthCheck.execute();
+            if (!result.isHealthy()) {
+                switch (healthCheck.getName()) {
```

Comment:
```suggestion
                healthCheckMetricProvider.measureUnhealthyCheck(result);
```
and overload `measureUnhealthyCheck` for 2 types of argument: `DatastoreQueryHealthCheck.Result` and `ThreadDeadlockHealthCheck.Result`. What do you think?